lib/registry/src/config.rs

```diff
@@ -0,0 +1,329 @@
+        }
+        match self {
+            Registries::Single(s) => s.url = registry,
+            Registries::Multi(m) => m.current = registry,
+        }
+    }
+
+    /// Returns the login token for the registry
+    pub fn get_login_token_for_registry(&self, registry: &str) -> Option<String> {
+        match self {
+            Registries::Single(s) if s.url == registry || format_graphql(registry) == s.url => {
+                s.token.clone()
+            }
+            Registries::Multi(m) => m
+                .tokens
+                .get(registry)
+                .or_else(|| m.tokens.get(&format_graphql(registry)))
+                .cloned(),
+            _ => None,
+        }
+    }
+
+    /// Sets the login token for the registry URL
+    pub fn set_login_token_for_registry(
+        &mut self,
+        registry: &str,
+        token: &str,
+        update_current_registry: UpdateRegistry,
+    ) {
+        let new_map = match self {
+            Registries::Single(s) => {
+                if s.url == registry {
+                    Registries::Single(Registry {
+                        url: format_graphql(registry),
+                        token: Some(token.to_string()),
+                    })
+                } else {
+                    let mut map = BTreeMap::new();
+                    if let Some(token) = s.token.clone() {
+                        map.insert(format_graphql(&s.url), token);
+                    }
+                    map.insert(format_graphql(registry), token.to_string());
+                    Registries::Multi(MultiRegistry {
+                        current: format_graphql(&s.url),
+                        tokens: map,
+                    })
+                }
+            }
+            Registries::Multi(m) => {
+                m.tokens.insert(format_graphql(registry), token.to_string());
+                if update_current_registry == UpdateRegistry::Update {
+                    m.current = format_graphql(registry);
+                }
+                Registries::Multi(m.clone())
+            }
+        };
+        *self = new_map;
+    }
+}
+
+impl PartialWapmConfig {
+    /// Save the config to a file
+    #[cfg(not(feature = "integration_tests"))]
+    pub fn save(&self) -> anyhow::Result<()> {
+        use std::{fs::File, io::Write};
+        let path = Self::get_file_location().map_err(|e| anyhow::anyhow!("{e}"))?;
+        let config_serialized = toml::to_string(&self)?;
+        let mut file = File::create(path)?;
+        file.write_all(config_serialized.as_bytes())?;
+        Ok(())
+    }
+
+    /// A mocked version of the standard function for integration tests
+    #[cfg(feature = "integration_tests")]
+    pub fn save(&self) -> anyhow::Result<()> {
+        let config_serialized = toml::to_string(&self)?;
+        crate::integration_tests::data::RAW_CONFIG_DATA.with(|rcd| {
+            *rcd.borrow_mut() = Some(config_serialized);
+        });
+
+        Ok(())
+    }
```

Comment:
Is this a good opportunity to remove the `feature = "integration_tests"` thing? We should be able to use the `tempfile` crate to get directories we can save to.

Reply:
ok done